includes/widgets/class-listings-widget.php

```diff
@@ -4,150 +4,214 @@
+		$default_image = $show_images && isset( $instance['default_image'] ) && $instance['default_image'];
+
+		$img_size = 'wpbdp-thumb';
+		if ( $show_images && ( $thumb_w > 0 || $thumb_h > 0 ) ) {
+			$img_size = array( $thumb_w, $thumb_h );
+		}
+
+		foreach ( $items as $post ) {
+			$html[] = $this->render_item( $post, $instance, $show_images, $default_image, $img_size, $html_class );
+		}
+
+		return join( "\n", $html );
+	}
+
+
+	private function get_item_thumbnail_position_css_class( $thumbnail_position, $version ) {
+		if ( $thumbnail_position == 'left' || $thumbnail_position == 'right' ) {
+			$css_class = sprintf( 'wpbdp-listings-widget-item-with-%s-thumbnail-in-%s', $thumbnail_position, $version );
+		} else {
+			$css_class = sprintf( 'wpbdp-listings-widget-item-with-thumbnail-above-in-%s', $version );
+		}
+
+		return $css_class;
+	}
+
+	private function render_item( $post, $instance, $show_images, $default_image, $img_size, $html_class ) {
+		$output        = '';
+		$listing       = wpbdp_get_listing( $post->ID );
+		$listing_title = sprintf( '<div class="wpbdp-listing-title"><a class="listing-title" href="%s">%s</a></div>', get_permalink( $post->ID ), get_the_title( $post->ID ) );
+		$html_image    = $this->render_image( $show_images, $default_image, $listing, $post, $img_size );
+
+		if ( ! empty( $html_image ) ) {
+			$template = '<li class="wpbdp-listings-widget-item %1$s"><div class="wpbdp-listings-widget-container wpbdp-clearfix"><div class="wpbdp-listings-widget-thumb wpbdp-clearfix">%2$s</div><div class="wpbdp-listings-widget-item--title-and-content">%3$s</div></div></li>';
+		} else {
+			$html_class .= ' wpbdp-listings-widget-item-without-thumbnail';
+			$template    = '<li class="wpbdp-listings-widget-item %1$s"><div class="wpbdp-listings-widget-container wpbdp-clearfix"><div class="wpbdp-listings-widget-item--title-and-content">%3$s</div></div></li>';
+		}
+
+		return sprintf( $template, $html_class, $html_image, $listing_title );
+	}
+
+	private function render_image( $show_images, $default_image, $listing, $post, $img_size ) {
+		$image_link = '';
+		if ( $show_images ) {
+			if ( $img_id = $listing->get_thumbnail_id() ) {
+				$image_link = '<a href="' . get_permalink( $post->ID ) . '">' . wp_get_attachment_image( $img_id, $img_size, false, array( 'class' => 'listing-image' ) ) . '</a>';
+			} elseif ( $default_image ) {
+				$size_class = $img_size;
+				$attribute  = '';
+				if ( is_array( $size_class ) ) {
+					$width      = $size_class[0] . 'px';
+					$height     = $size_class[1] . 'px';
+					$size_class = implode( 'x', $size_class );
+					$attribute  = "width='$width' height='$height'";
+				}
+				$class      = "attachment-$size_class size-$size_class listing-image";
+				$image_link = '<a href="' . get_permalink( $post->ID ) . '"><img src="' . WPBDP_ASSETS_URL . 'images/default-image-big.gif" class="' . $class . '" ' . $attribute . ' /></a>';
```

Comment:
Let's use the coming soon image here from the settings.

Reply:
Done, made changes to the helper class to make the function public